src/Form.jsx

```diff
@@ -131,11 +131,12 @@ var Form = React.createClass({
 
     // Does this field come with an associated label?
     if (label) {
-      label = <label key={name + 'label'} className={labelClass}>{label}</label>;
+      let optional = '';
       // mark optional fields that have a label as being optional:
       if (field.optional) {
-        label = [label, <span key={name + 'label-optional'} > (optional)</span>];
+        optional = <span key={name + 'label-optional'} className="optional">(optional)</span>;
       }
+      label = <label key={name + 'label'} className={labelClass}>{ label }{ optional }</label>;
```

Comment:
The previous code was weird, label could be either an element or an array of elements, and caused `CheckBox` to break because it passes the label's `props.children` on as its own content. That went horrible wrong with the optional span living "outside" of the label.